ksp_compiler3/ksp_compiler.py

```diff
@@ -1476,6 +1479,80 @@ def compress_variable_name(name):
 def default_read_file_func(filepath):
     return open(filepath, 'r').read()
 
+def parse_nckp(path):
+
+    '''
+    The prefix list is orderd to match the integer number representing each ui_control:
+
+    0. panel
+    1. button
+    2. fileselector
+    3. knob
+    4. label 
+    5. levelmeter
+    6. menu
+    7. slider
+    8. switch
+    9. table
+    10. textedit
+    11. valueedit
+    12. waveform
+    13. wavetable
+    14. xypad
+    15. mousearea
+
+    '''
+
+    prefix = ["$", "$", "$", "$", "$", "$", "$", "$", "$", "%", "@", "$", "$", "$", "?", "$"]
+    cur_prefix = []
+    
+    # if the nckp file is found, iterate recursively into the nested data                      
+    if os.path.exists(path):
+
+        def search_ui_in_dict_recursively(dictionary):
+            tree = ""
+            name = ""
+            key = 'index'
+            for k, v in dictionary.items():
+                if k == key:
+                    cur_prefix.append(prefix[v])
+                    tree = dictionary["value"]["common"]["id"]
+                    yield tree
+                elif isinstance(v, dict):
+                    for result in search_ui_in_dict_recursively(v):
+                        if tree:
+                            name = tree+"_"+result
```

Comment:
`name = (tree+"_"+result) if tree else result`

Don't know if it's more readable though.

Reply:
It is surely more pythonic, I agree

Reply:
A bit ugly (in my opinion) though. Any other opinions?

Reply:
The only thing that's maybe ugly in that line is that the concatenation doesn't have extra whitespaces for readability. :)

`name = (tree + "_" + result) if tree else result`

:)

Reply:
I can add the extra spaces without problem, but I like the one line solution provided

Reply:
Me too, it's readable enough.